poc/rfc7748.sage

```diff
@@ -0,0 +1,209 @@
+from sagelib.string_utils import *
+
+########## Definitions from RFC 7748 ##################
+
+def decodeLittleEndian(b, bits):
+    b_list = string_or_bytes_to_list(b)
+    return sum([b_list[i] << 8*i for i in range(floor((bits+7)/8))])
+
+def string_or_bytes_to_list(u):
+    try:
+        u_list = [ord(b) for b in u]
+    except:
+        u_list = [b for b in u]
+    return u_list
+
+def decodeUCoordinate(u, bits):
+    u_list = string_or_bytes_to_list(u)    
+    # Ignore any unused bits.
+    if bits % 8:
+        u_list[-1] &= (1<<(bits%8))-1
+    return decodeLittleEndian(u_list, bits)
+
+def encodeUCoordinate(u, bits):
+    num_bytes = floor((bits+7)/8)
+    result = bytearray(num_bytes)
+    for i in range(num_bytes):
+    	result[i] = (Integer(u) >> 8*i) & 0xff
+    return bytes(result)
+
+def decodeScalar25519(k):
+    k_list = string_or_bytes_to_list(k)    
+    k_list[0] &= 248
+    k_list[31] &= 127
+    k_list[31] |= 64
```

Comment:
Sorry @chris-wood, I think actually the previous configuration was correct. In RFC 7748 it is done correctly. If we can copy these lines over to the scalar generation code, we should be good.